### PR TITLE
Fix bug #78 - touch cache file on mac and linux

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: src/Microsoft.Identity.Client.Extensions.Msal/FileIOWithRetries.cs [hunk 1]
```diff
@@ -43,11 +55,32 @@ internal static void WriteDataToFile(string filePath, byte[] data, TraceSourceLo
...
+        /// </remarks>
+        internal static void TouchFile(string filePath, TraceSourceLogger logger)
+        {
+            string directoryForCacheFile = GetOrCreateParentDirectory(filePath, logger);
+            logger.LogInformation($"Cache file directory exists. '{Directory.Exists(directoryForCacheFile)}' now touching cache file");
 
             TryProcessFile(() =>
             {
-                File.WriteAllBytes(filePath, data);
+                if (!File.Exists(filePath))
```
Comment (rayluo) [inline src/Microsoft.Identity.Client.Extensions.Msal/FileIOWithRetries.cs:76]: :+1: Good catch on this edge case!

And then the `File.SetLastWriteTimeUtc(filePath, DateTime.UtcNow);` could probably be put into an `else` clause, the difference would be really minimal, though.

File: tests/Microsoft.Identity.Client.Extensions.Msal.UnitTests/TraceStringListener.cs [hunk 2]
```diff
@@ -3,11 +3,24 @@
...
 using System.Text;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
 
 namespace Microsoft.Identity.Client.Extensions.Msal.UnitTests
 {
     public class TraceStringListener : TraceListener
     {
+        private const string TraceSourceName = "TestSource";
+
+        public static (TraceSource, TraceStringListener) Create()
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Client.Extensions.Msal.UnitTests/TraceStringListener.cs:14]: Value tuples are pretty cool. :)

File: tests/Microsoft.Identity.Client.Extensions.Msal.UnitTests/TraceStringListener.cs [hunk 3]
```diff
@@ -3,11 +3,24 @@
...
+        {
+            var logger = new TraceSource(TraceSourceName, SourceLevels.All);
+            var listner = new TraceStringListener();
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Client.Extensions.Msal.UnitTests/TraceStringListener.cs:17]: typo: `listner` > `listener`
